### PR TITLE
[Remove Stream/Optional usage] LPS-171928 new_remove_stream_and_optional_in_saml_opensaml_integration - part 1

### PR DESCRIPTION
**Pull request by jeffwu0724**
Hi Tina, Shuyang,

As talked with Tina, I removed the use of AttributeValueWrapper in this PR, and only focusing on removing stream in DefaultAttributeResolver.java and AttributePublisherImpl.java
This is for https://issues.liferay.com/browse/LPS-171928.
Thanks for checking.

### CODE REVIEW

**Comment by jeffwu0724** [comment]
ci:test:saml

**Comment by Kvale1733** [comment]
@jeffwu0724 No regressions found.

File: modules/dxp/apps/saml/saml-opensaml-integration/src/main/java/com/liferay/saml/opensaml/integration/internal/resolver/AttributePublisherImpl.java [hunk 1]
```diff
@@ -44,40 +40,17 @@ public void publish(String name, String nameFormat, String... values) {
...
 
-		AttributeValue[] attributeValues = TransformUtil.transform(
-			values,
-			value -> new AttributeValueWrapper(
-				OpenSamlUtil.buildAttributeValue(value)),
-			AttributeValue.class);
+		List<XMLObject> xmlObjects = new ArrayList<>();
 
-		Stream<AttributeValue> stream = Arrays.stream(attributeValues);
+		for (String value : values) {
```
Comment (dantewang) [inline modules/dxp/apps/saml/saml-opensaml-integration/src/main/java/com/liferay/saml/opensaml/integration/internal/resolver/AttributePublisherImpl.java:45]: @jeffwu0724 We can also use `TransformUtil` here.

**Review by dantewang — CHANGES_REQUESTED** [review]
@jeffwu0724 Hi, could you explain a bit about the motivation of removing `AttributeValueWrapper`?

**Comment by jeffwu0724** [comment]
> @jeffwu0724 Hi, could you explain a bit about the motivation of removing `AttributeValueWrapper`?

Hi @dantewang , as discussed with Tina, `AttributeValueWrapper` was a private class, and only used to get the `xmlObject`. There is no need to make a wrapper with xmlobject, and then unwrap it to get the xmlobject. We can just pass in the value to make the xmlobject in `attributePublisher.publish`.  By removing `AttributeValueWrapper` , we can make transformUtil clearer from
```
TransformUtil.transformToArray(
						roles,
						role -> attributePublisher.buildString(role.getName()),
						AttributePublisher.AttributeValue.class));
```
to
```
TransformUtil.transformToArray(
						roles, Role::getName, String.class)
```
cc: @tinatian

**Comment by jeffwu0724** [comment]
ci:test:saml

**Comment by jeffwu0724** [comment]
ci:test:relevant

File: modules/dxp/apps/saml/saml-opensaml-integration/src/main/java/com/liferay/saml/opensaml/integration/internal/resolver/AttributePublisherImpl.java (hunk 1 above)
Comment (jeffwu0724) [reply modules/dxp/apps/saml/saml-opensaml-integration/src/main/java/com/liferay/saml/opensaml/integration/internal/resolver/AttributePublisherImpl.java]: changed to use transformUtil here

**Comment by Kvale1733** [comment]
@jeffwu0724 No regressions found.

**Comment by dantewang** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/128732

**Pull request closed without merging**